tests/contrib/vertica/test_vertica.py

```diff
@@ -0,0 +1,341 @@
+            }
+        }
+    )
+    def test_configuration_routine(self):
+        """Ensure that the integration routines can be configured."""
+        patch()
+        import vertica_python
+
+        test_tracer = get_dummy_tracer()
+
+        conn = vertica_python.connect(**VERTICA_CONFIG)
+        Pin.override(conn, tracer=test_tracer)
+        conn.cursor()  # should be traced now
+        conn.close()
+        spans = test_tracer.writer.pop()
+        assert len(spans) == 1
+        assert spans[0].name == "get_cursor"
+
+    def test_execute_metadata(self, test_conn, test_tracer):
+        """Metadata related to an `execute` call should be captured."""
+        conn, cur = test_conn
+
+        Pin.override(cur, tracer=test_tracer)
+
+        cur.execute("INSERT INTO {} (a, b) VALUES (1, 'aa');".format(TEST_TABLE))
+        cur.execute("SELECT * FROM {};".format(TEST_TABLE))
+        conn.close()
+
+        spans = test_tracer.writer.pop()
+        assert len(spans) == 2
+
+        # check all the metadata
+        assert spans[0].service == "vertica"
+        assert spans[0].span_type == "sql"
+        assert spans[0].name == "vertica.query"
+        assert spans[0].get_metric("db.rowcount") == -1
+        query = "INSERT INTO test_table (a, b) VALUES (1, 'aa');"
+        assert spans[0].get_tag("query") == query
+        assert spans[0].get_tag("out.host") == "127.0.0.1"
+        assert spans[0].get_tag("out.port") == "5433"
+        assert spans[0].get_tag("db.name") == "docker"
+        assert spans[0].get_tag("db.user") == "dbadmin"
+
+        assert spans[1].get_tag("query") == "SELECT * FROM test_table;"
+
+    def test_cursor_override(self, test_conn):
+        """Test overriding the tracer with our own."""
+        conn, cur = test_conn
+
+        test_tracer = get_dummy_tracer()
+        Pin.override(cur, tracer=test_tracer)
+
+        cur.execute("INSERT INTO {} (a, b) VALUES (1, 'aa');".format(TEST_TABLE))
+        cur.execute("SELECT * FROM {};".format(TEST_TABLE))
+        conn.close()
+
+        spans = test_tracer.writer.pop()
+        assert len(spans) == 2
+
+        # check all the metadata
+        assert spans[0].service == "vertica"
+        assert spans[0].span_type == "sql"
+        assert spans[0].name == "vertica.query"
+        assert spans[0].get_metric("db.rowcount") == -1
+        query = "INSERT INTO test_table (a, b) VALUES (1, 'aa');"
+        assert spans[0].get_tag("query") == query
+        assert spans[0].get_tag("out.host") == "127.0.0.1"
+        assert spans[0].get_tag("out.port") == "5433"
+
+        assert spans[1].get_tag("query") == "SELECT * FROM test_table;"
+
+    def test_execute_exception(self, test_conn, test_tracer):
+        """Exceptions should result in appropriate span tagging."""
+        from vertica_python.errors import VerticaSyntaxError
+
+        conn, cur = test_conn
+
+        Pin.override(cur, tracer=test_tracer)
+
+        with conn, pytest.raises(VerticaSyntaxError):
```

Comment:
neat, never new you could do `with item, item, item:`

Reply:
also, we are doing `with conn:` here, should we be doing that in our other tests as well?